src/pages/ChangePassword.ts

```diff
@@ -0,0 +1,119 @@
+import Component from '../utils/component';
+import ProfileForm from '../components/form/profile-form';
+import Input from '../components/input';
+import List from '../components/list';
+import LineInput from '../components/lineinput';
+import Button from '../components/button';
+import {REG_EXP, ERROR_MSG, Field, FieldBlock} from '../utils/validationConst';
+import LeftNav from '../components/leftnav';
+import Avatar from '../components/avatar';
+import { PATH } from '../router/paths';
+import userApi from '../api/user-api';
+import { User } from './ProfilePage';
+import Router from '../router';
+
+export default class ChangePassword extends List {
+	router = new Router('#root');
+	constructor(tag = 'div', props: {user?: User} = {}) {
+
+		const classAvatar = 'avatar';
+		const avatarProps = {};
+		const avatar: Avatar = new Avatar(
+			'div',
+			avatarProps,
+			classAvatar
+		);
+		
+		const fields: Field[] = [
+				{type: 'password', label: 'Старый\u00A0пароль', name: 'oldPassword', value: '', 
+				valid: REG_EXP.PASSWORD, errorMsg: ERROR_MSG.PASSWORD, autocomplete: 'current-password'},
+				{type: 'password', label: 'Новый\u00A0пароль', name: 'newPassword', value: '', 
+				valid: REG_EXP.PASSWORD, errorMsg: ERROR_MSG.PASSWORD},
+				{type: 'password', label: 'Повторите\u00A0новый\u00A0пароль', name: 'confirmPassword', 
+				value: '', valid: REG_EXP.PASSWORD, errorMsg: ERROR_MSG.CONFIRM_NEW_PASSWORD},
+				{type: 'hidden', name: 'login', value: props?.user?.login, autocomplete: 'login'},
+			];
```

Comment:
Так будет выглядеть читаемее. И я бы вынес эти константы из функции по аналогии с тем, как ты сделал с константами для валидации

```suggestion
				const fields: Field[] = [
			{
				type: 'password',
				label: 'Старый\u00A0пароль',
				name: 'oldPassword',
				value: '',
				valid: REG_EXP.PASSWORD,
				errorMsg: ERROR_MSG.PASSWORD,
				autocomplete: 'current-password'
			},
			{
				type: 'password',
				label: 'Новый\u00A0пароль',
				name: 'newPassword',
				value: '',
				valid: REG_EXP.PASSWORD,
				errorMsg: ERROR_MSG.PASSWORD
			},
			{
				type: 'password',
				label: 'Повторите\u00A0новый\u00A0пароль',
				name: 'confirmPassword',
				value: '',
				valid: REG_EXP.PASSWORD,
				errorMsg: ERROR_MSG.CONFIRM_NEW_PASSWORD
			},
			{
				type: 'hidden',
				name: 'login',
				value: props?.user?.login,
				autocomplete: 'login'
			},
		];
```

Reply:
Исправил